pkg/controller/servicebindingrequest/secret.go

```diff
@@ -61,8 +69,8 @@ func (s *secret) createOrUpdate(payload map[string][]byte) (*unstructured.Unstru
 
 // commit will store informed data as a secret, commit it against the API server. It can forward
 // errors from custom environment parser component, or from the API server itself.
-func (s *secret) commit(payload map[string][]byte) (*unstructured.Unstructured, error) {
-	return s.createOrUpdate(payload)
+func (s *secret) commit(payload map[string][]byte, secretOwnerReference v1.OwnerReference) (*unstructured.Unstructured, error) {
```

Comment:
@DhritiShikhar just to make code readable I feel we should have seperate method on type `secret`  which can be `setOwnerReference(reference)` instead of passing it to `commit` method.

Reply:
@akashshinde I made some changes here related to this here 3a5341c
Is this good?

Reply:
@DhritiShikhar Yes